packages/visualizations/src/components/MapPoi/Map.ts

```diff
@@ -122,26 +140,110 @@ export default class MapPOI {
              * We ask for features that are not in base style layers
              */
             const features = map.queryRenderedFeatures(point, { layers: this.layerIds });
-            this.updatePopup(map, features);
+            this.onFeaturesClick(map, features);
```

Comment:
It's just a slight readability suggestion, open for discussion:

First point, not mandatory at all, maybe rename this function onFeaturesClick to handleFeaturesAndPopups or something more explanatory of what the function do.

Second point, now as this function has the map instance we can probably pass the clicked `point` as an argument and place the` const features = map.queryRenderedFeatures(point, { layers: this.layerIds });` logic inside the function to make the onClick more pure ?